packages/backend/src/queryBuilder.ts

```diff
@@ -73,14 +73,14 @@ const renderMetricSql = (metric: Metric, explore: Explore): string => {
 }
 
 
-const renderDimensionSql = (dimension: Dimension, explore: Explore): string => {
+const renderDimensionSql = (dimension: Dimension, explore: Pick<Explore, 'tables'>): string => {
```

Comment:
just curious what the benefit to picking subfields out? Good practice?

Reply:
By having a more specific typing allows us to call this method without a complete explore object.
We could also change the method argument to be `tables: Table[]` and avoid mentioning the explore object.
In my case I'm calling this method while I'm generating an explore object.

Reply:
Since I have a conflict and I need to make another commit I'm going to make that change.

Reply:
Cool. 

I wonder if there's a way we can document / enforce. The dimension references are going to be limited to the tables passed into that function. 

The `Explore` kind of made it implicit that dimensions can only resolve within an explore (you can't reference a table in another explore).

Reply:
We can't trust the args. Even the `Explore` object could have been manipulated or poorly updated.  
We should make util functions throw errors if the necessary conditions aren't met. 
Having unit tests would also help.